q2_types/feature_data/_transformer.py

```diff
@@ -220,32 +220,16 @@ def _27(ff: BIOMV210Format) -> TSVTaxonomyFormat:
         table = biom.Table.from_hdf5(fh)
     return _biom_to_tsv_taxonomy_format(table)
 
-
-# DNA FASTA transformers
-
-class DNAIterator(collections.abc.Iterable):
-    def __init__(self, generator):
-        self.generator = generator
-
-    def __iter__(self):
-        yield from self.generator
+# common to all FASTA transformers
 
 
-class PairedDNAIterator(DNAIterator):
-    pass
-
+def _read_from_fasta(path, constructor=skbio.DNA):
+    return skbio.read(path, format='fasta', constructor=constructor)
 
-class AlignedDNAIterator(DNAIterator):
-    pass
 
-
-def _read_dna_fasta(path):
-    return skbio.read(path, format='fasta', constructor=skbio.DNA)
-
-
-def _dnafastaformats_to_series(ff):
+def _fastaformats_to_series(ff, constructor=skbio.DNA):
```

Comment:
Just a sidenote - this rename breaks q2-quality-control, but that's fine - q2-q-c is using private machinery here. I have opened a PR to fix this up:

https://github.com/qiime2/q2-quality-control/pull/58